### PR TITLE
Refactor deploy config and add support for S3 index strategy

### PR DESCRIPTION
**Pull request by ghedamat**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -26,146 +26,239 @@ function sshToServer(username, domain) {
...
+
+    let domain = this.pluginPackConfig.domain(this.deployTarget);
+    let username = this.pluginPackConfig.SSH_USERNAME;
+    let bastionHost = this.pluginPackConfig.bastionHost(this.deployTarget);
 
-      let domain = pluginPackConfig.domain(deployTarget);
-      let username = pluginPackConfig.SSH_USERNAME;
-      let bastionHost = pluginPackConfig.bastionHost(deployTarget);
+    return sshToServer(username, bastionHost || domain).then(function() {
+      require('dotenv').load({path: '.env.remote'});
```
Comment (jherdman) [inline index.js]: Does this need to be returned?

File: index.js [hunk 2]
```diff
@@ -26,146 +26,239 @@ function sshToServer(username, domain) {
...
+
+      this.slack();
+
+      this.sentry();
+
+      return this.ENV;
+    });
+  }
+
+  async getNoeConfig() {
```
Comment (jherdman) [inline index.js]: This doesn't appear to actually be async. Is that okay?

File: index.js (hunk 1 above)
Comment (ghedamat) [reply index.js]: nope, it's a shitty thing but it alters `process.env` instead

File: index.js (hunk 2 above)
Comment (ghedamat) [reply index.js]: you are right, it is async depending on the code branch, it should be fine

**Pull request merged**